### DISCUSSION TITLE
How to set up file transfers to my GitHub pages website?

### DISCUSSION THREAD

**Discussion by 3sv1n** [community/community#23489] [Other Feature Feedback, Questions, & Ideas]
<p>I need to set up an option to upload files to my website so I searched online and saw this tool <a href="https://filezilla-project.org/" rel="noopener nofollow ugc">FileZilla</a> I have downloaded this and when setting up my account I found out that I need the following to continue setup:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/0/90159eccd5ff0a992c490f38e69aa7288a76e461.jpeg" data-download-href="/uploads/short-url/kyD4MbAmE6OJdariQVXWyY5Dssh.jpeg?dl=1" title="FTPS error" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/72260184/181077375-410063a0-b934-4806-8816-70c21394ba4b.jpeg" alt="FTPS error" data-base62-sha1="kyD4MbAmE6OJdariQVXWyY5Dssh" width="690" height="132" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/0/90159eccd5ff0a992c490f38e69aa7288a76e461_2_690x132.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/9/0/90159eccd5ff0a992c490f38e69aa7288a76e461_2_1035x198.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/3X/9/0/90159eccd5ff0a992c490f38e69aa7288a76e461.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/0/90159eccd5ff0a992c490f38e69aa7288a76e461_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">FTPS error</span><span class="informations">1153×222 54.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>please help me guys, I need to create this soon.</p>

**Comment by jfullstackdev** [comment]
<p>you mean you have your files for the GitHub Page and you want to transfer that on a GitHub repo ? even a drag-and-drop will do,</p>

**Comment by 3sv1n** [comment]
<p>No I mean I need to enable file upload to my website, I don’t need drag-and-drop for a GitHub repo, I do that with GitHub Desktop.</p>

**Comment by jfullstackdev** [comment]
<p>you want to use FTP to transfer files ?</p>
<p>where is the server ? is it a GitHub repo that was turned as GitHub Pages ?</p>
<p>if you have a separate server, then I think you have the complete details of the instruction on how to move files at that server,</p>
<p>in the past, if I can still remember it correctly, I tried Microsoft IIS, once you setup the server locally, you still need authentication and authorization for the FTP service, so you can move files on the server and you need additional setup if you want to make that available over the Internet, not just on the LAN</p>

**Comment by yoannchaudet** [comment]
<p><a class="mention" href="/devjoshua312">@devjoshua312</a> GitHub Pages does not support the FTP protocol. You can upload files directly from <a href="http://github.com">github.com</a> to your repository though. <a href="https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site#creating-your-site" class="inline-onebox">Creating a GitHub Pages site - GitHub Docs</a> may help you too.</p>
<p>If you want to be able to upload files directly from your Pages site unfortunately, this is not something we provide safe mechanisms for.</p>

**Comment by 3sv1n** [comment]
<p>oh. thank you for the clarification <a class="mention" href="/yoannchaudet">@yoannchaudet</a> . Just one more thing, Can I add a FTP to my website to upload file to my system? like, a visitor of my webpage clicks a link which enable him to select files and upload them to my local machhine?</p>

**Comment by Simran-B** [comment]
<p>Well, yes, but it doesn’t have anything to do with GitHub Pages. You will need to build an upload form, run some server software on your local machine that can handle file uploads, use a DynDNS service to give your computer a domain name that automatically updates the DNS record with your IP address, and potentially leave your computer running 24/7 to make uploads work all the time.</p>
